pkg/agent/containerd/containerd.go

```diff
@@ -101,17 +101,17 @@ func Run(ctx context.Context, cfg *config.Node) error {
 		os.Exit(1)
 	}()
 
-	if err := WaitForContainerd(ctx, cfg.Containerd.Address); err != nil {
+	if err := WaitForCRIService(ctx, cfg.Containerd.Address, "containerd"); err != nil {
 		return err
 	}
 
 	return preloadImages(ctx, cfg)
 }
 
-// WaitForContainerd blocks in a retry loop until the Containerd CRI service
+// WaitForCRIService blocks in a retry loop until the CRI service
 // is functional at the provided socket address. It will return only on success,
 // or when the context is cancelled.
-func WaitForContainerd(ctx context.Context, address string) error {
+func WaitForCRIService(ctx context.Context, address string, service string) error {
```

Comment:
Should this maybe get moved out to the utils package now that it's not containerd-specific? `pkg/util/cri.go` or something?

Reply:
Consolidated them under `/pkg/cri` (/pkg/agent/util was causing k3s to explode in size to 89MB)